base/file.jl

```diff
@@ -230,23 +230,27 @@ julia> mkpath("intermediate_dir/actually_a_directory.txt") # creates two directo
 julia> isdir("intermediate_dir/actually_a_directory.txt")
 true
 
+julia> mkpath("my/test/dir/") # returns the original `path`
+"my/test/dir/"
 ```
 """
 function mkpath(path::AbstractString; mode::Integer = 0o777)
-    isdirpath(path) && (path = dirname(path))
     dir = dirname(path)
+    # stop recursion for `""`, `"/"`, or existed dir
     (path == dir || isdir(path)) && return path
     mkpath(dir, mode = checkmode(mode))
     try
-        mkdir(path, mode = mode)
+        # cases like `mkpath("x/")` will cause an error if `isdir(path)` is skipped
+        # the error will not be rethrowed, but it may be slower, and thus we avoid it in advance
+        isdir(path) || mkdir(path, mode = mode)
     catch err
         # If there is a problem with making the directory, but the directory
         # does in fact exist, then ignore the error. Else re-throw it.
         if !isa(err, IOError) || !isdir(path)
             rethrow()
         end
     end
-    path
+    return path
```

Comment:
```suggestion
    path
```
Not idiomatic

Reply:
Both styles are used in the codes, and it might be clearer to return this data more explicitly.